Solution Operator role manifest

alexandre-allard opened the pull request:
**Component**: salt, solutions, examples

**Context**: 

**Summary**:

**Acceptance criteria**: 

Waiting for #2565 to be merged.

---

Closes: #2389

alexandre-allard committed:
salt: add method to get roles manifest from Solution ISO

This method returns the Role & ClusterRole to create
for a Solution Operator.
It checks if the provided kinds are allowed and also
appends the Solution namespace to the namespaced objects.

Refs: #2389

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2389-solution-provide-role-manifest` into
`development/2.5`
* [w/2.6/improvement/2389-solution-provide-role-manifest](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2389-solution-provide-role-manifest) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

bert-e commented:
# History mismatch



Merge commit #eb81c1fe968ff64a53107810e3546d24e8bc3d7a on the integration branch
`w/2.6/improvement/2389-solution-provide-role-manifest` is merging a branch which is neither the current
branch `improvement/2389-solution-provide-role-manifest` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2389-solution-provide-role-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet reviewed (commented):
LGTM, maybe just add details in the Solution Operator guidelines for how this will work (especially the error case if something else than `Role`/`ClusterRole` objects are defined in the file). Would make sense to provide an easy tool for validation of an ISO (outside of a running MetalK8s cluster).

alexandre-allard committed:
doc: warn user about Solution role.yaml allowed content

Refs: #2389

bert-e commented:
# History mismatch



Merge commit #1dc0c2c30f5f049763c625b6c9341bc64dea526a on the integration branch
`w/2.6/improvement/2389-solution-provide-role-manifest` is merging a branch which is neither the current
branch `improvement/2389-solution-provide-role-manifest` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2389-solution-provide-role-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard committed:
salt: deploy roles from Solution role.yaml manifest

It allows Solution to declare what Roles and
CusterRoles are needed for the Operator to work.
Then it creates the ServiceAccount and binds
(RoleBinding) all the roles to this account.

Refs: #2389

alexandre-allard committed:
salt: remove old role.yaml manifest

This file is no longer used since we now
let the user define its own roles.

Refs: #2389

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2389-solution-provide-role-manifest` into
`development/2.5`
* [w/2.6/improvement/2389-solution-provide-role-manifest](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2389-solution-provide-role-manifest) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

gdemonet commented on docs/developer/solutions/operator.rst:
```diff
@@ -105,3 +105,7 @@ directory. This file is a manifest which declares all necessary ``Role`` and
 named ``<solution_name>-operator`` and all needed ``RoleBinding`` to bind these
 roles to this account.
+
+.. warning:: Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
+             the deployment of the Solution fail if any other resource is
+             found.
```
Would be best written like:

```suggestion
.. warning::

   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
   the deployment of the Solution fail if any other resource is found.
```

gdemonet commented on salt/metalk8s/orchestrate/solutions/files/operator/role_binding.yaml:
```diff
@@ -1,7 +1,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: {{ solution }}-operator
+  name: {{ solution }}-operator-{{ role_kind }}-{{ role_name }}
```
```suggestion
  name: {{ solution }}-operator-{{ role_kind | lower }}-{{ role_name }}
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #aa0fe84bd28417fdfde9e6bbd6cede78241fa885 on the integration branch
`w/2.6/improvement/2389-solution-provide-role-manifest` is merging a branch which is neither the current
branch `improvement/2389-solution-provide-role-manifest` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2389-solution-provide-role-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2389-solution-provide-role-manifest` into
`development/2.5`
* [w/2.6/improvement/2389-solution-provide-role-manifest](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2389-solution-provide-role-manifest) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #57f7b0c64637804d921d92214cb7a067b61b75a1 on the integration branch
`w/2.6/improvement/2389-solution-provide-role-manifest` is merging a branch which is neither the current
branch `improvement/2389-solution-provide-role-manifest` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2389-solution-provide-role-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2389-solution-provide-role-manifest` into
`development/2.5`
* [w/2.6/improvement/2389-solution-provide-role-manifest](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2389-solution-provide-role-manifest) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

Ebaneck commented on docs/developer/solutions/operator.rst:
```diff
@@ -105,3 +105,8 @@ directory. This file is a manifest which declares all necessary ``Role`` and
+.. warning::
+
+   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
+   the deployment of the Solution fail if any other resource is found.
```
Minor
```suggestion
   Only ``Role`` and ``ClusterRole`` kinds are allowed in this file,
   the deployment of the Solution fails if any other resource is found.
```

bert-e commented:
# History mismatch



Merge commit #6c027317d1a46c9fb88da97b63244f653feb04d8 on the integration branch
`w/2.6/improvement/2389-solution-provide-role-manifest` is merging a branch which is neither the current
branch `improvement/2389-solution-provide-role-manifest` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/2389-solution-provide-role-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2389-solution-provide-role-manifest` into
`development/2.5`
* [w/2.6/improvement/2389-solution-provide-role-manifest](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/2389-solution-provide-role-manifest) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

alexandre-allard commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality










*The following options are set:* **approve**

alexandre-allard committed:
example: add pyyaml dependency for manifest script

Refs: #2389

Ebaneck reviewed (approved):
Couldn't successfully run Post-merge on this!
Let's merge it.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
